### ISSUE TITLE
Error message too vague: "You may need an appropriate loader to handle this file type."

### ISSUE THREAD

**Issue by nhooey** [webpack/webpack#9025]
# Bug report

**What is the current behavior?**

When running `npx webpack`, I get the following vague error message:

```
ERROR in ./src/App.tsx 6:16
Module parse failed: Unexpected token (6:16)
You may need an appropriate loader to handle this file type.
```

**If the current behavior is a bug, please provide the steps to reproduce.**

1. Run this:
```
npx create-react-app example --typescript
```

2. Use this `webpack.config.js` file:
```javascript
const path = require('path');

module.exports = {
    // change to .tsx if necessary
    entry: './src/App.tsx',
    output: {
        filename: 'bundle.js',
        path: path.resolve(__dirname, 'dist')
    },
    resolve: {
        // changed from extensions: [".js", ".jsx"]
        extensions: ['.ts', '.tsx', '.js', '.jsx']
    },
    module: {
        rules: [
            // changed from { test: /\.jsx?$/, use: { loader: 'babel-loader' } },
            { test: /\.(t|j)sx?$/, use: { loader: 'awesome-typescript-loader' } },
            // addition - add source-map support
            { enforce: "pre", test: /\.js$/, loader: "source-map-loader" }
        ]
    },
    externals: {
        "react": "React",
        "react-dom": "ReactDOM",
    },
    // addition - add source-map support
    devtool: "source-map"
}
```

**What is the expected behavior?**

The error message should mention what loader, if any or not, was used to attempt to load the file. The message does not currently definitively say what is configured.

In verbose mode, the `module: { rules: ... }` section should also mention when it's applying a rule to a file that matched a pattern, or when a file didn't match any patterns. Currently the verbose output doesn't print enough information to see what's going on.

**Other relevant information:**

webpack version:  `4.29.6`
Node.js version: `11.13.0`
Operating System: Mac OS 10.14.4
Additional tools: `awesome-typescript-loader 5.2.1`

**Comment by evilebottnawi** [comment]
Feel free to send a PR

**Comment by cacheflow** [comment]
I've always wanted this as well or for Webpack to recommend a possible loader based on the file extension. I'm happy to do the first part of this and display what loaders were used when trying to load the file.

**Comment by evilebottnawi** [comment]
@cacheflow :+1:

**Comment by cacheflow** [comment]
Here's a PR for this. 

https://github.com/webpack/webpack/pull/9150

**Comment by nhooey** [comment]
@cacheflow Awesome, thanks!

It's great when someone fixes vague error messages. You'll save new users a lot of time.

**Comment by srikanthsunkari** [comment]
When can this be available? Please publish the #9150 PR.
